metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -91,12 +92,16 @@ def extract_dbt_entities(
     load_catalog: bool,
     target_platform: str,
     environment: str,
+    node_filter: list,
```

Comment:
Could we use the same "AllowDenyPattern" class that we use in a couple other places? I think it'd be more flexible and consistent.

Also it would be good to have a default for this field to allow everything so we don't break backwards compatibility. 

```suggestion
    node_filter: AllowDenyPattern = AllowDenyPattern.allow_all(),
```